src/dft.cpp

```diff
@@ -1171,6 +1182,8 @@ complex<double> fields::process_dft_component(dft_chunk **chunklists, int num_ch
 /***************************************************************/
 complex<realnum> *collapse_array(complex<realnum> *array, int *rank, size_t dims[3], direction dirs[3], volume where);
 
+size_t *reduce_array_dimensions(volume where, int full_rank, int &reduced_rank, size_t &reduced_grid_size, size_t dims[3], size_t stride[3], size_t reduced_stride[3], direction dirs[3], direction reduced_dirs[3]);
```

Comment:
Probably both of these should be in `meep-internal.hpp` … and make sure that header file is included both here and in `array_slice.cpp` where the functions are defined.

The reason is that this way the compiler will check that the declaration in the header matches the declaration where the function is defined.